### ISSUE TITLE
TypeError: Cannot read property '1' of null

### ISSUE THREAD

**Issue by rzmz** [benwinding/react-admin-firebase#129]
Hi,

When using this in a component provided to react-admin through customRoutes with noLayout set to true, there is an error thrown to console:
```
index.js:2178 TypeError: Cannot read property '1' of null
    at DataProvider.ts:87
```

It occurs with the following code (first create a user with the correct id):
```
  const { data  } = useQueryWithStore({
    type: "getOne",
    resource: "users",
    payload: { id: "ciixOthYIWeS17Pafsj4" },
  });
```

I get the same result when either running this authenticated and with an id not existing in the users collection or when not authenticated. I think the better way would be to handle potential firebase errors correctly.

Firebase returned error when not authenticated:
code: "permission-denied"
message: "Missing or insufficient permissions."

I believe the culprit is in status-code-translator.ts:33.

Or is it a bug in react-admin? Or am I doing something wrong?

**Comment by rzmz** [comment]
My rules are nothing special:

```
rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      allow read, write: if request.auth.uid != null;
    }
  }
}
```

**Comment by benwinding** [comment]
Thanks for reporting the issue @rzmz,

> I believe the culprit is in status-code-translator.ts:33.

Looks like you're right, it seems to be a problem with the `regex` parser for the `status-code-translator`.

https://github.com/benwinding/react-admin-firebase/blob/2b5e17dbd098307c80f64c8d4eed51028a0b3de0/src/misc/status-code-translator.ts#L33

This should be deployed in `v3.2.1`.

Try it out and let me know if the error persists or if there's anymore error messages.

Cheers,
Ben

**Comment by benwinding** [comment]
This also might be related to #123 which is caused by having a `DocumentReference` field in a document. This will be fixed soon by PR #127
